test/integration/RedisResourceLockerIntegration.test.ts

```diff
@@ -0,0 +1,218 @@
+import type { RedisResourceLocker } from '../../src';
+import { joinFilePath, readableToString } from '../../src';
+import type { HttpServerFactory } from '../../src/server/HttpServerFactory';
+import { describeIf } from '../util/TestHelpers';
+import { instantiateFromConfig } from './Config';
+/**
+ * Test the general functionality of the server using a RedisResourceLocker
+ */
+describeIf('docker', 'A server with a RedisResourceLocker as ResourceLocker', (): void => {
+  const port = 6008;
+  const baseUrl = `http://localhost:${port}/`;
+  let server: Server;
+  let locker: RedisResourceLocker;
+  let factory: HttpServerFactory;
+
+  beforeAll(async(): Promise<void> => {
+    const instances = await instantiateFromConfig(
+      'urn:solid-server:test:Instances',
+      'run-with-redlock.json',
+      {
+        'urn:solid-server:default:variable:baseUrl': baseUrl,
+        'urn:solid-server:default:variable:podTemplateFolder': joinFilePath(__dirname, '../assets/templates'),
+      },
+    ) as Record<string, any>;
+    ({ factory, locker } = instances);
+    server = factory.startServer(port);
+  });
+
+  afterAll(async(): Promise<void> => {
+    await locker.quit();
+    await new Promise<void>((resolve, reject): void => {
+      server.close((error): void => error ? reject(error) : resolve());
+    });
+  });
+
+  it('can add a file to the store, read it and delete it.', async(): Promise<void> => {
+    // Create file
+    const fileUrl = `${baseUrl}testfile2.txt`;
+    const fileData = await fs.readFile(
+      joinFilePath(__dirname, '../assets/testfile2.txt'),
+    );
+
+    let response = await fetch(fileUrl, {
+      method: 'PUT',
+      headers: {
+        'content-type': 'text/plain',
+      },
+      body: fileData,
+    });
+    expect(response.status).toBe(205);
+
+    // Get file
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(200);
+    const body = await readableToString(response.body as any);
+    expect(body).toContain('TESTFILE2');
+
+    // DELETE file
+    response = await fetch(fileUrl, {
+      method: 'DELETE',
+    });
+    expect(response.status).toBe(205);
+    response = await fetch(fileUrl);
+    expect(response.status).toBe(404);
+  });
+
+  it('can create a folder and delete it.', async(): Promise<void> => {
+    const containerPath = 'secondfolder/';
+    const containerUrl = `${baseUrl}${containerPath}`;
+    // PUT
+    let response = await fetch(containerUrl, {
+      method: 'PUT',
+      headers: {
+        'content-type': 'text/plain',
+      },
+    });
+    expect(response.status).toBe(205);
+
+    // GET
+    response = await fetch(containerUrl);
+    expect(response.status).toBe(200);
+
+    // DELETE
+    response = await fetch(containerUrl, {
+      method: 'DELETE',
+    });
+    expect(response.status).toBe(205);
+    response = await fetch(containerUrl);
+    expect(response.status).toBe(404);
+  });
+
+  it('can upload and delete an image.', async(): Promise<void> => {
```

Comment:
Would just remove this test. The redis locker does not care about the difference between an image and text body, and the second one is already tested above. Otherwise you would probably still need to keep the fs import just for reading that image.